File: frontend/src/api/admin/GetDirectorPermissionsApi.ts
import { GetDirectorPermissionsRequestQuerySchema, GetDirectorPermissionsResponseDataSchema, type DirectorPermissionSummary } from "../api-contracts.js";
import { request } from "../client.js";

export const GetDirectorPermissionsApiPath = "/admin/director/permissions" as const;

export const getDirectorPermissions = async (userId: string): Promise<DirectorPermissionSummary> => {
  GetDirectorPermissionsRequestQuerySchema.parse({});
  return request(GetDirectorPermissionsApiPath, { method: "GET", headers: { "x-user-id": userId } }, GetDirectorPermissionsResponseDataSchema);
};

File: backend/microservice/src/admin/director/api/GetDirectorPermissionsApi.scala
package microservice.admin.director.api

import cats.effect.IO
import java.sql.Connection
import microservice.admin.director.objects.DirectorPermissionSummary
import microservice.core.{APIWithTokenMessage, AccessControl, HttpError}
import microservice.system.objects.AdminLevel

final case class GetDirectorPermissionsAPIMessage(
  userId: String
) extends APIWithTokenMessage[DirectorPermissionSummary] {
  override def token: String = userId

  override def plan(connection: Connection): IO[Either[HttpError, DirectorPermissionSummary]] =
    IO.pure {
      AccessControl.requireAdminLevel(connection, userId, AdminLevel.Director).map { user =>
        DirectorPermissionSummary(
          userId = user.id,
          adminLevel = AdminLevel.Director,
          canManageUiCustomization = true
        )
      }
    }
}

object GetDirectorPermissionsEndpoint {
  val name: String = "GetDirectorPermissions"
  val method: String = "GET"
  val path: String = "/admin/director/permissions"
  val businessLogic: String =
    "总监管理员权限探测接口，用于验证当前 admin 账号是否具备 director 权限。"
}
